src/formatDate.js

```diff
@@ -1,14 +1,41 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // split date by 4 position('-') to new array;
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {}; // create an empty object
+  const changedDate = [];
+
+  // check each part of date(YY or YYYY) and saving year value to receivedDate;
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        receivedDate['YY'] = dateArr[i].slice(2);
+        receivedDate['YYYY'] = dateArr[i];
+        break;
+      case 'YY':
+        receivedDate['YYYY'] = `${dateArr[i] < 30 ? 20 : 19}${dateArr[i]}`;
+        receivedDate['YY'] = dateArr[i];
+        break;
+      default:
+        receivedDate[fromFormat[i]] = dateArr[i];
+        break;
+    }
+  }
+
+  // add parts of final date to new array with rules of toFormat;
+  for (const elem of toFormat.slice(0, -1)) {
+    changedDate.push(receivedDate[elem]);
+  }
+
+  return changedDate.join(toFormat[3]);
```

Comment:
Similar to the issue in line 12, the delimiter is assumed to be at the fourth position in the `toFormat` array. Ensure that the delimiter is correctly identified and used for joining the date parts.